app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -171,9 +171,10 @@ def unlock
 
       def visit
         @page.unlock!
-        redirect_to show_page_path(
+        redirect_to show_page_url(
           urlname: @page.urlname,
-          locale: prefix_locale? ? @page.language_code : nil
+          locale: prefix_locale? ? @page.language_code : nil,
+          host: @page.site.host
```

Comment:
Not all sites have a valid hostname, ie. [the default site Alchemy seeds](https://github.com/AlchemyCMS/alchemy_cms/blob/master/config/alchemy/config.yml#L83).